src/dex/aave-v3-stata/aave-v3-stata.ts

```diff
@@ -0,0 +1,507 @@
+  //   srcAmount: string,
+  //   destAmount: string,
+  //   data: AaveV3StataData,
+  //   side: SwapSide,
+  // ): Promise<SimpleExchangeParam> {
+  //   const { exchange, srcType, destType } = data;
+  //   let swapData;
+  //
+  //   if (side === SwapSide.SELL) {
+  //     if (srcType === TokenType.STATA_TOKEN) {
+  //       // e.g. sell srcAmount 100 srcToken stataUSDC for destToken USDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.redeem, [
+  //         srcAmount,
+  //         this.augustusAddress, // receiver
+  //         this.augustusAddress, // owner
+  //         destType === TokenType.UNDERLYING, // withdraw from aToken
+  //       ]);
+  //     } else {
+  //       // sell srcAmount 100 srcToken USDC for destToken stataUSDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(
+  //         StataFunctions.deposit,
+  //         [
+  //           srcAmount,
+  //           this.augustusAddress, // receiver
+  //           0, // referrer (noop)
+  //           srcType === TokenType.UNDERLYING, // deposit to aave
+  //         ],
+  //       );
+  //     }
+  //   } else {
+  //     if (srcType === TokenType.STATA_TOKEN) {
+  //       // e.g. buy destAmount 100 destToken USDC for srcToken stataUSDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(
+  //         StataFunctions.withdraw,
+  //         [
+  //           destAmount,
+  //           this.augustusAddress, // receiver
+  //           this.augustusAddress, // owner
+  //         ],
+  //       );
+  //     } else {
+  //       // e.g. buy destAmount 100 destToken stataUSDC for srcToken USDC
+  //       swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.mint, [
+  //         destAmount,
+  //         this.augustusAddress,
+  //       ]);
+  //     }
+  //   }
+  //
+  //   return this.buildSimpleParamWithoutWETHConversion(
+  //     srcToken,
+  //     srcAmount,
+  //     destToken,
+  //     destAmount,
+  //     swapData,
+  //     exchange,
+  //   );
+  // }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: AaveV3StataData,
+    side: SwapSide,
+    _: Context,
+    executorAddress: Address,
+  ): DexExchangeParam {
+    const { exchange, srcType, destType } = data;
+    let swapData;
+    let returnAmountPos = undefined;
+
+    if (side === SwapSide.SELL) {
+      if (srcType === TokenType.STATA_TOKEN) {
+        // e.g. sell srcAmount 100 srcToken stataUSDC for destToken USDC
+        swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.redeem, [
+          srcAmount,
+          recipient, // receiver
+          executorAddress, // owner
+          destType === TokenType.UNDERLYING, // withdraw from aToken
+        ]);
+
+        returnAmountPos = extractReturnAmountPosition(
+          AaveV3Stata.stata,
+          StataFunctions.redeem,
+          '',
+          1,
```

Comment:
Internal type is `uint256` not `uint256[]`, so no need to extract dynamic params

Reply:
We get 2 values back [here](https://github.com/paraswap/paraswap-dex-lib/pull/700/files#diff-1deec285bd07ad57d19bd1f7a4a229997802c9b4c2fddd2d901412fb32b1e5b3R659-R661)

Reply:
This param works if returned type is array. For now this function returns tuple. Then we should adapt function to handle tuples, not just arrays

Reply:
Ok, as only a tuple is returned, then you can use `RETURN_AMOUNT_POS_32` instead of interacting with `extractReturnAmountPosition`

If you need to get first element, not second, from the tuple, then you should use `RETURN_AMOUNT_POS_0`

Reply:
I'll leave it undefined for this case, and leave a TODO to improve, would need to retest thoroughly

Reply:
No need to postpone this, it's an easy update.

Just use return_amount_pos_32 as const value if you want to extract second uint256 value from static calldata.

calladata encoding [link](https://medium.com/@scourgedev/deep-dive-into-abi-encode-types-padding-and-disassembly-84472f1b4543)

Reply:
done